Realm/RLMQueryUtil.mm

```diff
@@ -1881,3 +1892,30 @@ bool is_self_value_for_key_path_function_expression(NSExpression *expression)
                     @"Property '%@' not found in object of type '%@'", columnName, desc.className);
     return prop;
 }
+
+void RLMGetPathElements(std::vector<PathElement> &paths, NSExpression *expression) {
```

Comment:
I don't see any reason for the paths not to just be a return value here.

It is ver non-obvious what is going on in this function and why, and it needs some comments explaining.

Reply:
Added comments and add it to the internal namespace